autoprotocol/builders.py

```diff
@@ -1557,38 +1576,127 @@ def device_mode_params(
+                if key == "nozzle_size" or key == "z_drop":
+                    device_dict.update({key: Unit(value)})
+                else:
+                    device_dict.update({key: value})
+
+        return {device: device_dict}
+
+    @staticmethod
+    def validate_device_params(device: str, device_dict: dict) -> None:
+        """
+        Helper validation function to validate device liquid handling params
+        Parameters
+        ----------
+        device: str
+            either x_mantis or x_tempest_chip
+        device_dict: dict
+            Dictionary of all device params, as seen in function device_mode_params
+
+        Raises
+        ------
+        ValueError
+            If input device is not either x_mantis or x_tempest_chip
+        ValueError
+            If input device_dict values are not of the accepted
+            params for the input device
+        KeyError
+            If input device_dict key is not in accepted_params keys
+        """
+        if device == "x_mantis":
+            # If device is mantis, set accepted params
+            accepted_params: dict = {
+                "model": ["high_volume", "low_volume"],
+                "diaphragm": [0, 100],
+                "nozzle_size": [Unit("0.1:mm"), Unit("0.2:mm"), Unit("0.5:mm")],
+                "tubing": ["LV", "HV", "P200", "P1000"],
+                "z_drop": [Unit("0.0:mm"), Unit("100.0:mm")],
+                "viscosity": ["1", "2-5", "6-10", "11-20", "21-25"],
+            }
+        elif device == "x_tempest_chip":
+            # Otherwise, default to tempest accepted params
+            accepted_params: dict = {
+                "model": ["high_volume"],
+                "nozzle": ["standard"],
+                "material": ["pfe", "silicone"],
             }
-            device_mode_params["x_tempest_chip"] = x_tempest_chip
+        else:
+            raise ValueError(
+                f"Device is {device}. It must be: [x_tempest_chip, x_mantis]"
+            )
+        error_values: dict = {}
+        error_keys: dict = {}
+        # Validate params with accepted params dict
+        for key, value in device_dict.items():
+            if value:
+                if key not in accepted_params:
+                    error_keys.update({key: value})
+                elif key == "diaphragm":
+                    accepted_range: List = list(accepted_params[key])
+                    if value < accepted_range[0] or value > accepted_range[1]:
+                        error_values.update({key: value})
+                elif key == "z_drop":
+                    accepted_range: List = list(accepted_params[key])
+                    if isinstance(value, str):
+                        compare_val = Unit(value)
+                    else:
+                        compare_val = value
+                    if (
+                        compare_val < accepted_range[0]
+                        or compare_val > accepted_range[1]
+                    ):
```

Comment:
i _think_ you could write this like

```suggestion
                    if accepted_range[0] > compare_val > accepted_range[1]:
```